pulumi/folio/index.ts

```diff
@@ -104,6 +111,69 @@ const folioDeployment = new FolioDeployment(tenant,
 // Create a configMap for folio for certain non-secret environment variables that will be deployed.
 const appName = "folio";
 const appLabels = { appClass: appName };
+
+// Create a secret for folio to store our environment variables that k8s will inject into each pod.
+// These secrets have been set in the stack using the pulumi command line.
+const config = new pulumi.Config();
+
+const inClusterDbHost = config.requireSecret("db-host");
+const dbAdminUser = config.requireSecret("db-admin-user");
+const dbAdminPassword = config.requireSecret("db-admin-password");
+const dbUserName = config.requireSecret("db-user-name");
+const dbUserPassword = config.requireSecret("db-user-password");
+
+// TODO Add tags to this and other new stuff.
+// TODO Make this dependent on the vpc.
+const dbSubnetGroup = new aws.rds.SubnetGroup("folio-db-subnet", {
+    subnetIds: folioVpc.privateSubnetIds
+});
+export const pgFinalSnapshotId = "folio-pg-cluster-final-snapshot";
+export const pgClusterId = "folio-pg-cluster";
+const clusterName = "folio-pg";
+const pgCluster = new aws.rds.Cluster(clusterName, {
+    // TODO Be careful with this list. It seems to error out if 3 items aren't provided on
+    // subsequent runs doing a replace because of a diff on availabilityZones. The error is:
+    // error creating RDS cluster: DBClusterAlreadyExistsFault: DB Cluster already exists.
+    // But I'm not seeing that it can't create in us-west-2c, because I believe we only have
+    // two azs in the vpc. Making it 2 seems to work again. Ok well it will attempt
+    // to replace when there are only 2 still. Adding 3 makes it not do that, which is
+    // probably wrong since the vpc only has 2.
+
+    // I believe destroying and creating the stack with 3 azs in the vpc should fix this.
+    availabilityZones: [
+        "us-west-2a",
+        "us-west-2b",
+        "us-west-2c"
+    ],
+    backupRetentionPeriod: 5,
+    clusterIdentifier: pgClusterId,
+    databaseName: "postgres",
+    engine: "aurora-postgresql",
+    masterPassword: pulumi.interpolate`${dbAdminPassword}`,
+    masterUsername: pulumi.interpolate`${dbAdminUser}`,
+    preferredBackupWindow: "07:00-09:00",
```

Comment:
Is this UTC? If so, this is probably okay. If it is timezone specific, we should target off hours

Reply:
It is UTC. 7:00 AM UTC is 1:00AM Central.

Reply:
This is all pg 12.7 on Aurora. That's what pulumi `rds.Custer` and `rds.ClusterInstance` is. See: https://www.pulumi.com/registry/packages/aws/api-docs/rds/cluster/.

Reply:
Should we explicitly declare the version of postgresql that we want to prevent unexpected upgrades/updates

Reply:
Yes the `engineVersion` can be specified as a property of `rds.Cluster` and also as a property of `ClusterInstance`. I'm trying it with `12.7` for both. I recall seeing something in the docs somewhere about suppressing automatic minor version upgrades. I think part of the appeal of Aurora is that they're taking care of patches for you but I think you have some control too.